include/external/mpark/config.hpp

```diff
@@ -0,0 +1,79 @@
+// MPark.Variant
```

Comment:
Should/could we treat this (and related) as an external sub-module rather the distributing with BOUT++?

Reply:
I think the advantage is that it might make it easier to get updates should there be bug fixes etc.  rather than manually merging changes (and it perhaps makes it clearer that we shouldn't touch these files, although that's indicated by being in externals) but the downside is that an initial checkout of BOUT++ will still need a network connection at the configure stage.

Reply:
I'd prefer to stick it as a submodule under `externalpackages/`